api/src/main/java/io/opentelemetry/correlationcontext/CorrelationsContextUtils.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * @since 0.1.0
+ */
+@Immutable
+public final class CorrelationsContextUtils {
+  private static final Context.Key<CorrelationContext> CORR_CONTEXT_KEY =
+      Context.key("opentelemetry-corr-context-key");
+
+  /**
+   * Creates a new {@code Context} with the given value set.
+   *
+   * @param corrContext the value to be set.
+   * @return a new context with the given value set.
+   * @since 0.1.0
+   */
+  public static Context withCorrelationContext(CorrelationContext corrContext) {
+    return withCorrelationContext(corrContext, Context.current());
+  }
+
+  /**
+   * Creates a new {@code Context} with the given value set.
+   *
+   * @param corrContext the value to be set.
+   * @param context the parent {@code Context}.
+   * @return a new context with the given value set.
+   * @since 0.1.0
+   */
+  public static Context withCorrelationContext(CorrelationContext corrContext, Context context) {
+    return context.withValue(CORR_CONTEXT_KEY, corrContext);
+  }
+
+  /**
+   * Returns the {@link CorrelationContext} from the current {@code Context}, falling back to a
+   * default, no-op {@link CorrelationContext}.
+   *
+   * @return the {@link CorrelationContext} from the current {@code Context}.
+   * @since 0.3.0
+   */
+  public static CorrelationContext getCorrelationContext() {
+    return getCorrelationContext(Context.current());
+  }
+
+  /**
+   * Returns the {@link CorrelationContext} from the specified {@code Context}, falling back to a
+   * default, no-op {@link CorrelationContext}.
+   *
+   * @param context the specified {@code Context}.
+   * @return the {@link CorrelationContext} from the specified {@code Context}.
+   * @since 0.3.0
+   */
+  public static CorrelationContext getCorrelationContext(Context context) {
+    CorrelationContext corrContext = CORR_CONTEXT_KEY.get(context);
+    return corrContext == null ? EmptyCorrelationContext.getInstance() : corrContext;
+  }
+
+  /**
+   * Returns the {@link CorrelationContext} from the specified {@code Context}. If none is found,
+   * this method returns {code null}.
+   *
+   * @param context the specified {@code Context}.
+   * @return the {@link CorrelationContext} from the specified {@code Context}.
+   * @since 0.1.0
+   */
+  @Nullable
+  public static CorrelationContext getCorrelationContextWithoutDefault(Context context) {
+    return CORR_CONTEXT_KEY.get(context);
+  }
+
+  /**
+   * Returns a new {@link Scope} encapsulating the provided {@link CorrelationContext} added to the
+   * current {@code Context}.
+   *
+   * @param corrContext the {@link CorrelationContext} to be added to the current {@code Context}.
+   * @return the {@link Scope} for the updated {@code Context}.
+   * @since 0.1.0
+   */
+  public static Scope withScopedCorrelationContext(CorrelationContext corrContext) {
+    Context context = withCorrelationContext(corrContext);
+    return ContextUtils.withScopedContext(context);
+  }
```

Comment:
This needs to signal that it interacts with the "Current" Context. Maybe rename:
`currentContextWith(CorrelationContext corrContext)`

Reply:
Will follow this up in another PR.